### ISSUE TITLE
[2017.5] Several distros failing integration.states.test_pip test on Python 3

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt-jenkins#338]
The `integration.states.test_pip.PipStateTest.test_pip_installed_weird_install` is failing on the [2017.5 Python 3](https://jenkins.saltstack.com/job/branch_tests/job/2017.5/view/Python%203/) tests for several distros:
```
Traceback (most recent call last):
  File "//testing/tests/support/helpers.py", line 964, in decorator
    return func(cls, grains=cls.run_function('grains.items'))
  File "/testing/tests/integration/states/test_pip.py", line 126, in test_pip_installed_weird_install
    self.assertTrue(ret[key]['result'])
AssertionError: False is not true
```

This test is failing on the following distros:
- CentOS 7
- SUSE LEAP
- Ubuntu 16.04
- Ubuntu 17.04

There is also a pip test failing in the same file in Debian 8, but the out put is different. This is documented in #337.

**Comment by terminalmage** [comment]
Can't reproduce this.

**Comment by rallytime** [comment]
@terminalmage Just running the single test? Or is it possible that other tests are interfering? I am unsure what the situation is with these PY 3 tests in that regard, especially since this has failed on so many distros lately.

**Comment by terminalmage** [comment]
Yeah, I was only running a single test or single test module. While investigating #340, I found that a different test was modifying ``os.environ['SHELL']``, which seemed to be a potential cause of #340. Perhaps that ``os.environ`` change caused some of these other failures as well?

This is also confounded by the fact that the failures do not consistently show up in the test results.

https://github.com/saltstack/salt/pull/41308 fixes the ``os.environ`` issue, perhaps we should wait until that is merged and see if these go away? Though again, I'm not sure how we'd tell if they went away when they aren't consistently failing in the first place.

**Comment by rallytime** [comment]
@terminalmage I think this test is being affected by other tests or something else going on. It's failing pretty consistently across the distros, so other tests (the whole file, all of the state tests (`-S` option) or the module tests maybe (`-m`) are interfering somehow. 

However, I noticed this is a duplicate of #337. I'll let you and @garethgreenaway decide which issue to close and who will chase the rest of this.

I was hoping https://github.com/saltstack/salt/pull/41308 would fix this also, but since that was merged several days ago, I think something is still going on here.
